test/byContract/IexecPoco/04_finalize.test.ts

```diff
@@ -356,6 +356,135 @@ describe('IexecPoco2#finalize', async () => {
+        {
+            testInfo: 'Get at least MIN_KITTY if available',
+            workerpoolPriceToFillKitty: 3_333_333_334,
+            expectedKittyBeforeFinalize: 1_000_000_000, // 1 RLC
+            expectedSchedulerKittyPartReward: 1_000_000_000,
+        },
+        {
+            testInfo: 'Get all kitty when the kitty is lower than MIN_KITTY',
+            workerpoolPriceToFillKitty: 3_333_333_330,
+            expectedKittyBeforeFinalize: 999_999_999, // ~0.99 RLC
+            expectedSchedulerKittyPartReward: 999_999_999,
+        },
+    ].forEach((testArgs) => {
+        const {
+            testInfo,
+            workerpoolPriceToFillKitty,
+            expectedKittyBeforeFinalize,
+            expectedSchedulerKittyPartReward,
+        } = testArgs;
+        it(`Should finalize task with scheduler kitty part reward (${testInfo})`, async () => {
+            // Fill kitty
+            const kittyAddress = await iexecPoco.kitty_address();
+            const kittyFillingDealVolume = 1;
+            const kittyFillingSchedulerDealStake = await iexecWrapper.computeSchedulerDealStake(
+                workerpoolPriceToFillKitty,
+                kittyFillingDealVolume,
+            );
+            const kittyFillingSchedulerTaskStake =
+                kittyFillingSchedulerDealStake / kittyFillingDealVolume;
+            const kittyFilling = await iexecWrapper.signAndMatchOrders(
+                ...buildOrders({
+                    assets: ordersAssets,
+                    requester: requester.address,
+                    prices: {
+                        app: 0,
+                        dataset: 0,
+                        workerpool: workerpoolPriceToFillKitty, // 30% will go to kitty
+                    },
+                    volume: kittyFillingDealVolume,
+                    salt: ethers.utils.id(new Date().toISOString()),
+                }).toArray(),
+            );
+            await iexecPoco
+                .initialize(kittyFilling.dealId, kittyFilling.taskIndex)
+                .then((tx) => tx.wait());
+            const kittyFrozenBeforeClaim = (await iexecPoco.frozenOf(kittyAddress)).toNumber();
+            const task = await iexecPoco.viewTask(kittyFilling.taskId);
+            await time.setNextBlockTimestamp(task.finalDeadline);
+            await expect(iexecPoco.claim(kittyFilling.taskId))
+                .to.emit(iexecPoco, 'Transfer')
+                .withArgs(iexecPoco.address, kittyAddress, kittyFillingSchedulerTaskStake)
+                .to.emit(iexecPoco, 'Reward')
+                .withArgs(kittyAddress, kittyFillingSchedulerTaskStake, kittyFilling.taskId)
+                .to.emit(iexecPoco, 'Transfer')
+                .withArgs(kittyAddress, iexecPoco.address, kittyFillingSchedulerTaskStake)
+                .to.emit(iexecPoco, 'Lock')
+                .withArgs(kittyAddress, kittyFillingSchedulerTaskStake)
+                .to.changeEtherBalances([iexecPoco, kittyAddress], [0, 0]);
```

Comment:
why ether balances and not changeTokenBalances ?

Reply:
Thanks a lot!